client/src/pages/UserSettings.tsx

```diff
@@ -47,18 +50,56 @@ const LanguageButtonFin = styled(Button)(() => ({
   border: '2px solid Gray',
   color: 'black',
 }))
+
 const UserSettings = () => {
   // TODO: @finnan - get all the details from db
+  interface User {
+    username: string
+  }
+
+  const [userSub, setUserSub] = useState<string>('')
+  const [userData, setUserData] = useState<UserBySubQuery | null>(null)
+
+  const user = useAuthUser() as User | null
+  
+  useEffect(() => {
+    if (user && user.username) {
+      setUserSub(user.username)
+    }
+  }, [user])
+
+  const fetchedUserData = useGetUserData(userSub)
+
+  useEffect(() => {
+    if(fetchedUserData) {
+      setUserData(fetchedUserData)
+    }
+  }, [fetchedUserData])
+
   const initialUserSetting = {
     firstName: '',
     lastName: '',
     email: '',
     selectedLanguage: 'English',
-    interests: [],
+    interests:  [] as (string | null)[],
   }
 
   const [formData, setFormData] = useState(initialUserSetting)
 
+  useEffect(() => {
+    const userItem = userData?.userBySub?.items[0]
+    if (userItem) {
+      const { givenName, familyName, email, language, interestTags } = userItem;
+      setFormData({
```

Comment:
We probably don't need to use setFormData here. For the initial stage of the user settings page, we can add the data we get from `userData` and replace the values in `initialUserSetting`. `setFormData` should only be updated by the input or setting changes done by the user

Reply:
We can also possibly minimize the amount of useEffects